### PR TITLE
support storage.sync

### PR DESCRIPTION
**Pull request by joehillen**
After having to reset all the settings on all my machines because of the new FF57 update, I was thinking it would be nice if TST supported sync. So I decided to try and implement https://github.com/piroor/treestyletab/issues/1078 This PR is the first step.

### CODE REVIEW

File: Configs.js [hunk 1]
```diff
@@ -72,6 +73,11 @@ Configs.prototype = {
         this.$log(`load: try load from storage on  ${location.href}`);
         values = await browser.storage.local.get(this.$default);
         values = values || this.$default;
+        if (this.$syncKeys && this.$syncKeys.length) {
+          syncedValues = await browser.storage.sync.get(this.$syncKeys);
```
Comment (piroor) [inline Configs.js]: Is this new local variable? then `let` seems to be required.

**Comment by piroor** [comment]
Thanks! I've added a comment.

**Comment by piroor** [comment]
And, could you add descriptions how to use the sync feature to the README.md?

**Comment by joehillen** [comment]
Fixed with a force push

**Comment by piroor** [comment]
Thanks. One more point I forgot to comment; this library is designed to start loading automatically when an `Configs` instance is created. So I think we need to specify sync keys as a part of arguments for the constructor.

For example, from:

```javascript
configs = new Configs({
  style: '',
  defaultStyle: 'mixed',
  defaultStyleOnDarwin: 'sidebar',
  defaultStyleOnLinux: 'plain',
...
});
```

to:

```javascript
configs = new Configs({
  style: '',
  defaultStyle: 'mixed',
  defaultStyleOnDarwin: 'sidebar',
  defaultStyleOnLinux: 'plain',
...
}, {
  syncKeys: ['style', 'defaultStyle', ...]
});
```

**Comment by joehillen** [comment]
Is this what you have in mind?

```javascript
function Configs(aDefaults, args = {syncKeys: []}) {
...
  this.$syncKeys = args.syncKeys;
}
```

**Comment by piroor** [comment]
Oh, yes! After that I'm ready to merge this.

**Comment by joehillen** [comment]
Fixed with a force push

**Comment by piroor** [comment]
Thanks a lot！

**Pull request merged**